frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,877 @@
+		/// - `child_bounty_id`: Index of child bounty.
+		#[pallet::weight(<T as Config>::WeightInfo::unassign_curator())]
+		pub fn unassign_curator(
+			origin: OriginFor<T>,
+			#[pallet::compact] parent_bounty_id: BountyIndex,
+			#[pallet::compact] child_bounty_id: BountyIndex,
+		) -> DispatchResult {
+			let maybe_sender = ensure_signed(origin.clone())
+				.map(Some)
+				.or_else(|_| T::RejectOrigin::ensure_origin(origin).map(|_| None))?;
+
+			// Ensure parent bounty exist, get active status info.
+			let (parent_curator, update_due) = Self::ensure_bounty_active(parent_bounty_id)?;
+
+			// Ensure child-bounty is in expected state.
+			ChildBounties::<T>::try_mutate_exists(
+				parent_bounty_id,
+				child_bounty_id,
+				|maybe_child_bounty| -> DispatchResult {
+					let mut child_bounty =
+						maybe_child_bounty.as_mut().ok_or(BountiesError::<T>::InvalidIndex)?;
+
+					let slash_curator = |curator: &T::AccountId,
+					                     curator_deposit: &mut BalanceOf<T>| {
+						let imbalance = T::Currency::slash_reserved(curator, *curator_deposit).0;
+						T::OnSlash::on_unbalanced(imbalance);
+						*curator_deposit = Zero::zero();
+					};
+
+					match child_bounty.status {
+						ChildBountyStatus::Added => {
+							// No curator to unassign at this point.
+							return Err(BountiesError::<T>::UnexpectedStatus.into())
+						},
+						ChildBountyStatus::CuratorProposed { ref curator } => {
+							// A child-bounty curator has been proposed, but not accepted yet.
+							// Either `RejectOrigin`, parent-bounty curator or the proposed
+							// child-bounty curator can unassign the child-bounty curator.
+							ensure!(
+								maybe_sender.map_or(true, |sender| sender == *curator ||
+									sender == parent_curator),
+								BadOrigin,
+							);
+						},
+						ChildBountyStatus::Active { ref curator } => {
+							// The child-bounty is active.
+							match maybe_sender {
+								// If the `RejectOrigin` is calling this function,
+								// slash the curator deposit.
+								None => {
+									slash_curator(curator, &mut child_bounty.curator_deposit);
+									// Continue to change child-bounty status below...
+								},
+								Some(sender) => {
+									if sender == *curator {
+										// This is the child-bounty curator,
+										// willingly giving up their role.
+										// Give back their deposit.
+										T::Currency::unreserve(
+											&curator,
+											child_bounty.curator_deposit,
+										);
+										// Reset curator deposit.
+										child_bounty.curator_deposit = Zero::zero();
+									// Continue to change bounty status below...
+									} else if parent_curator == sender {
+										// Looks like child-bounty curator is inactive,
+										// slash their deposit.
+										slash_curator(curator, &mut child_bounty.curator_deposit);
+									// Continue to change child-bounty status below...
+									} else {
+										// Check for expiry,
```

Comment:
I won't complain, but I do recommend _not_ formatting your code comments in multiple lines for no reason. This impairs readability in my opinion (and make it look a bit  like a poem :D).

Reply:
Sure, I have run rewrap on these ones as well. Some of them are wrapped short because of the indentation. But with rewrap at least they are all similarly formatted. Thanks.